### PR TITLE
Switch to gunicorn for serving of python models.

### PR DESCRIPTION
**Pull request by tomasatdatabricks**
## What changes are proposed in this pull request?
PyFunc serve now uses gunicorn instead of just flask. This makes it more useful out of the box.
 
## How is this patch tested?
Add tests covering the use of new gunicorn arguments added to the cli. Other than that the existing tests should suffice.
 
## Release Notes
The serving of python models now uses gunicorn as its webserver.  The use of gunicorn makes the serving functionality production ready out of the box. 

### Is this a user-facing change? 

- [ ] No. You can skip the rest of this section.
- [x] Yes. Give a description of this change to be included in the release notes for MLflow users.
 
(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)
 
### What component(s) does this PR affect?
 
- [ ] UI
- [x] CLI 
- [ ] API 
- [ ] REST-API 
- [ ] Examples 
- [ ] Docs
- [ ] Tracking
- [ ] Projects 
- [ ] Artifacts 
- [ ] Models 
- [ ] Scoring 
- [x] Serving
- [ ] R
- [ ] Java
- [x] Python

### How should the PR be classified in the release notes? Choose one:
 
- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [x] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: mlflow/pyfunc/backend.py [hunk 1]
```diff
@@ -19,8 +19,9 @@ class PyFuncBackend(FlavorBackend):
     """
         Flavor backend implementation for the generic python models.
     """
-    def __init__(self, config, no_conda=False, install_mlflow=False, **kwargs):
+    def __init__(self, config, workers=1, no_conda=False, install_mlflow=False, **kwargs):
```
Comment (smurching) [inline mlflow/pyfunc/backend.py]: Why is the default (one) here different from in the `models` CLI (four)? There's probably a good reason, I was just wondering

File: tests/models/test_cli.py [hunk 2]
```diff
@@ -116,6 +119,29 @@ def test_model_with_no_deployable_flavors_fails_pollitely():
         assert "No suitable flavor backend was found for the model." in stderr
 
 
+def test_serve_gunicorn_opts(iris_data, sk_model):
+    if sys.platform == "win32":
+        pytest.skip("This test requires gunicorn which is not available on windows.")
```
Comment (smurching) [inline tests/models/test_cli.py]: Hm, this is a good point, do we need to document that `mlflow models serve` doesn't yet work on Windows?

File: mlflow/pyfunc/backend.py (hunk 1 above)
Comment (smurching) [reply mlflow/pyfunc/backend.py]: Actually I think we might not need the default at all here?

**Review by smurching — COMMENTED** [review]
Thanks @tomasatdatabricks :) had a small question + a question about Windows support, otherwise LGTM

File: tests/models/test_cli.py (hunk 2 above)
Comment (smurching) [reply tests/models/test_cli.py]: From offline discussion, sounds like we can add Windows support in the future

Comment (tomasatdatabricks) [reply tests/models/test_cli.py]: As we discussed offline, we've never officially supported windows. The current level of support for windows only covers tracking.

**Pull request merged**